injector/src/test/java/edu/ucr/cs/riple/injector/ClassSearchTest.java

```diff
@@ -468,4 +475,32 @@ public void class_search_benchmark_22() {
                 true))
         .start(true);
   }
+
+  @Test
+  public void proper_report_for_class_not_found_test() {
```

Comment:
Actually I wanted to divide it into two steps, but that requires hardcoding the expected error message. Which, if we change the message, we need to update the tests as well. However, it makes more sense to divide it to two cases in favor of cohesiveness of each test.

9d5f3188f1500967ef879e7979ee0cc87b751adc

Please let me know if you prefer the older version.

Reply:
Hardcoding the expected message in tests and having to change the tests when behavior changes is very reasonable. In production code, I'd err on the side of not hardcoding things and avoiding repetition, since that makes the code fragile. But within unit tests, clarity is more important, and I'd argue that failing when existing behavior changes is a feature, not a bug, of a comprehensive test suite :)